src/compiler/crystal/syntax/parser.cr

```diff
@@ -1327,23 +1327,24 @@ module Crystal
     end
 
     def parse_begin
+      begin_location = @token.location
       slash_is_regex!
       next_token_skip_statement_end
       exps = parse_expressions
-      node, end_location = parse_exception_handler exps
+      node, end_location = parse_exception_handler exps, begin_location: begin_location
       node.end_location = end_location
       if !node.is_a?(ExceptionHandler) && (!node.is_a?(Expressions) || node.keyword)
-        node = Expressions.new([node]).at(node).at_end(node)
+        node = Expressions.new([node]).at(begin_location).at_end(end_location)
```

Comment:
I presume this change shouldn't be necessary. `at` and `at_end` take the start and end locations from `node` which should already be set appropriately.

Reply:
This is needed for:

```crystal
begin
  foo
end
```

because `node` is the AST node for `foo`